matlab/@Net/eval.m

```diff
@@ -164,6 +165,19 @@ function eval(net, inputs, mode, derOutput, accumulateParamDers)
           vars{inputDer} = vars{inputDer} + slice_der(args{:}) ;
         end
       end
+      
+      % remove derivatives (even input vars)
+      % as they are no longer used in the network
+    if conserveMemory
+        for i = numel(layer.inputVars):-1:1
```

Comment:
Maybe this loop can be vectorized? Suggestion (assuming I understand the logic correctly) :
`idx = (mod(layer.inputVars, 2) == 0) ;  % logical vector with 1 where inputVars must be deleted
vars(layer.inputVars(idx)) = [] ;  % delete all at once`

Reply:
Sure - it's now vectored - but using the syntax
` vars(layer.inputVars(idx)) = {[]};`
other wise MATLAB deletes the cell entry.